CHAOS-6976 Windows network latency (#8406)
harness/developer-hub: docs/chaos-engineering/use-harness-ce/chaos-faults/windows/windows-network-latency.md

---
id: windows-network-latency
title: Windows Network Latency
---

> Windows Network Latency simulates a network latency scenario on Windows VMs. It checks the performance of the application running on the Windows VMs.

Reviewer: by causing network packet delay using clumsy

Author: Where are we providing path of clumsy? If its just downloaded, is it sufficient?

Reviewer: At the time of download we are exporting the path so do not need to use the relative path.

![Windows Network Latency](./static/images/windows-network-latency.png)

## Use cases
- Determines the resilience of an application when a network delay scenario is simulated on a Windows virtual machine.
- Simulates the situation of network delay for processes running on the application, which degrades their performance.
- Helps verify the application's ability to handle network failures and its failover mechanisms.

## Prerequisites
- Ensure that the [prerequisites](/docs/chaos-engineering/use-harness-ce/chaos-faults/windows/prerequisites) are fulfilled before executing the experiment.

### Mandatory tunables

   <table>
      <tr>
        <th> Tunable </th>
        <th> Description </th>
        <th> Notes </th>
      </tr>
      <tr>
        <td> NETWORK_LATENCY </td>

>         <td> The network delay that you want to introduce during chaos, in milliseconds. </td>

Reviewer: ```suggestion
        <td> The network latency that you want to introduce during chaos, in milliseconds. </td>
```

        <td> For example, 2000. </td>
      </tr>
      <tr>
          <td> DESTINATION_HOSTS </td>
          <td> DNS or FQDN names of services whose access is affected. </td>
          <td> You can specify multiple inputs as comma-separated values. For example, "abc.com,github.com".</td>
      </tr>
    </table>

### Optional tunables
   <table>
      <tr>
        <th> Tunable </th>
        <th> Description </th>
        <th> Notes </th>
      </tr>
      <tr>
        <td> DESTINATION_IPS </td>
        <td> IP addresses of services (or pods) or CIDR block whose access is affected. </td>
        <td> </td>

>       </tr>

Reviewer: It is one off either host name or IP addresses, in the diagram also we have one more section other than `Mandatory` and `Optional` called `OneOf`

```yaml
apiVersion: litmuschaos.io/v1alpha1
kind: MachineChaosExperiment
metadata:
  name: windows-network-latency
spec:
  engineState: "active"
  chaosServiceAccount: litmus-admin
  experiments:
    infraType: windows
    steps:
      - - name: windows-network-latency
    tasks:
    - definition:
        chaos:
          env:
            - name: NETWORK_LATENCY
              value: "2000"
            - name: DESTINATION_HOSTS
              value: aws.amazon.com,github.com
```

### Destination hosts

The `DESTINATION_HOSTS` environment variable specifies the destination hosts to induce latency on the target Windows VM.

Use the following example to specify destination hosts:

[embedmd]:# (./static/manifests/windows-network-latency/network-latency.yaml yaml)
```yaml
apiVersion: litmuschaos.io/v1alpha1
kind: MachineChaosExperiment
metadata:
  name: windows-network-latency
spec:
  engineState: "active"
  chaosServiceAccount: litmus-admin
  experiments:
    infraType: windows
    steps:
      - - name: windows-network-latency
    tasks:
    - definition:
        chaos:
          env:
            - name: NETWORK_LATENCY
              value: "2000"

>             - name: DESTINATION_HOSTS

Reviewer: We can give an example of destination IP as well

---
id: windows-network-latency
title: Windows Network Latency
---

> Windows Network Latency causes a network packet delay on Windows VMs for the target hosts by causing network packet delay using clumsy. It checks the performance of the application running on the Windows VMs.

Reviewer: ```suggestion
Windows Network Latency causes a network packet delay on Windows VMs for the target hosts using Clumsy. It can also check the application's performance when running on Windows VMs.
```
